client/src/components/form/SignUp.js

```diff
@@ -0,0 +1,149 @@
+import logo from '../../logo192.png';
+import {useState} from 'react';
+import {useHistory, Link} from 'react-router-dom';
+import Form from 'react-bootstrap/Form';
+import Container from 'react-bootstrap/Container';
+import SubmitButton from 'react-bootstrap/Button';
+import Row from 'react-bootstrap/Row';
+import Col from 'react-bootstrap/Col';
+import {makeStyles} from '@material-ui/core';
+import Typography from '@material-ui/core/Typography';
+import Divider from '@material-ui/core/Divider';
+import MediaLogo from '@material-ui/core/Button';
+import FacebookIcon from '@material-ui/icons/Facebook';
+import MailIcon from '@material-ui/icons/Mail';
+import TwitterIcon from '@material-ui/icons/Twitter';
+import styles from '../../css/signUp.module.css';
+
+
+// media tabs
+const useStyles = makeStyles ({
+    gmail: {
+        backgroundColor: '#db4437',
+        '&:hover': {
+            backgroundColor: '#e2695f',
+        }
+    },
+
+    fb: {
+        backgroundColor: '#4267b2',
+        '&:hover': {
+            backgroundColor:  '#6785c1',
+        }
+    },
+
+    twitter: {
+        backgroundColor: '#00acee',
+        '&:hover': {
+            backgroundColor: '#33bcf1',
+        }
+    }
+})
```

Comment:
I have a question -- why can't we put this in a css and use styles.fb or styles.gmail or so to style the different respective components? I tried but it doesn't work the same way, and I can't figure out why...isn't styling in here and writing these classes in a css filed the same thing?

Reply:
It is a good question that requires few lines to explain:

For stylesheets, there are few common ways to do that:
- CSS
- Sass
- Less
- CSS in Javascript / [JSS](https://cssinjs.org/?v=v10.7.1)

For `import styles from './styles.css'`, you are actually using CSS and if you try to print something like `styles.author`, you will get a string **className** like `reviewList_author__26HZg` which is a auto generated class name base on your css file. (with some namespace and hash to make it unique)

For `makeStyles`, you are using CSS in Javascript / JSS, which aims to represent styles in JSON object, and convert to CSS   right before the moment you use it. So the output of `classes.fb` is a string like `makeStyles-fb-4` which is a object key to access JSON object representing the style.

So when you pass `styles.author` to material-ui components, the meaning of the string is different and they cannot recognize, so it cannot work.

You can check this link for why they are using JSS: https://material-ui.com/styles/basics/#why-use-material-uis-styling-solution

Reply:
ohhh, I see! Now that explains a lot -- that is also probably why I can't change the color of the material-ui components...Thank you!